parsons/databases/postgres/postgres_core.py

```diff
@@ -0,0 +1,242 @@
+                                connect_timeout=self.timeout)
+        yield conn
+
+        conn.commit()
+        conn.close()
+
+    @contextmanager
+    def cursor(self, connection):
+        cur = connection.cursor(cursor_factory=psycopg2.extras.DictCursor)
+        yield cur
+        cur.close()
+
+    def query(self, sql, parameters=None):
+        """
+        Execute a query against the database. Will return ``None``if the query returns zero rows.
+
+        To include python variables in your query, it is recommended to pass them as parameters,
+        following the `psycopg style <http://initd.org/psycopg/docs/usage.html#passing-parameters-to-sql-queries>`_.
+        Using the ``parameters`` argument ensures that values are escaped properly, and avoids SQL
+        injection attacks.
+
+        **Parameter Examples**
+
+        .. code-block:: python
+
+            # Note that the name contains a quote, which could break your query if not escaped
+            # properly.
+            name = "Beatrice O'Brady"
+            sql = "SELECT * FROM my_table WHERE name = %s"
+            rs.query(sql, parameters=[name])
+
+        .. code-block:: python
+
+            names = ["Allen Smith", "Beatrice O'Brady", "Cathy Thompson"]
+            placeholders = ', '.join('%s' for item in names)
+            sql = f"SELECT * FROM my_table WHERE name IN ({placeholders})"
+            rs.query(sql, parameters=names)
+
+        `Args:`
+            sql: str
+                A valid SQL statement
+            parameters: list
+                A list of python variables to be converted into SQL values in your query
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+
+        """  # noqa: E501
+
+        with self.connection() as connection:
+            return self.query_with_connection(sql, connection, parameters=parameters)
+
+    def query_with_connection(self, sql, connection, parameters=None, commit=True):
+        """
+        Execute a query against the database, with an existing connection. Useful for batching
+        queries together. Will return ``None`` if the query returns zero rows.
+
+        `Args:`
+            sql: str
+                A valid SQL statement
+            connection: obj
+                A connection object obtained from ``redshift.connection()``
+            parameters: list
+                A list of python variables to be converted into SQL values in your query
+            commit: boolean
+                Whether to commit the transaction immediately. If ``False`` the transaction will
+                be committed when the connection goes out of scope and is closed (or you can
+                commit manually with ``connection.commit()``).
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        # To Do: Have it return an ordered dict to return the
```

Comment:
I think `OrderedDict` is about returning dictionaries where the columns are in the right order. The query should return the rows in the right order because the list's order should be stable.

Reply:
Removed.